### PR TITLE
Introduce config schema shapes with defaults/constraints/transforms

### PR DESCRIPTION
**Pull request by delano**
Establishes work for #3212.

## Summary

Establishes a separation of concerns between type-only contracts and runtime-augmented shapes for configuration schemas. Introduces the `augment` helper utility to apply defaults and value constraints on top of contract schemas without restating their structure, then uses it to build shape variants for all major config sections.

### Key Changes

**New Utility:**
- `src/schemas/utils/augment.ts` — Helper function that recursively applies leaf transforms and sub-tree overrides to Zod object contracts. Handles unwrapping/rewrapping of `.optional()`, `.nullable()`, and `.default()` wrappers, preserves object-level qualifiers (`.catchall()`, `z.looseObject`), and validates unknown keys at construction time.

**New Shape Modules:**
- `src/schemas/shapes/config/` — Parallel structure to contracts with augmented schemas for all sections:
  - `billing.ts` — Restores description length bounds, limit value constraints (≥ -1), price amount bounds (≥ 0), plan display order (≥ 0), and match_fields default
  - `public.ts` — Applies TTL bounds, passphrase length limits (4–128), password-generation bounds, and default TTL (7 days)
  - `logging.ts` — Per-logger level defaults and slow-request threshold
  - `auth.ts` — Auth mode default, simple-mode positive bounds, database URL fallback
  - `config.ts` — Composes per-section shapes into runtime/static/mutable/legacy config schemas
  - Section shapes: `site.ts`, `mail.ts`, `jobs.ts`, `features.ts`, `storage.ts`, `jurisdiction.ts`, `diagnostics.ts`, `development.ts`, `i18n.ts`, `ui.ts`, `secret_options.ts`, `limits.ts`, `capabilities.ts`

**Contract Updates:**
- Removed `.default()` calls from all contract schemas (site, mail, features, storage, development, ui, diagnostics, i18n, auth, logging, public, secret_options, billing)
- Added documentation headers clarifying that contracts describe field names/types only; defaults and constraints live in shapes

**Test Coverage:**
- `src/tests/schemas/utils/augment.spec.ts` — 231 lines covering leaf overrides, sub-tree recursion, wrapper unwrap/rewrap, catchall preservation, and runtime unknown-key validation
- `src/tests/schemas/shapes/config/billing.spec.ts` — Smoke tests for billing shape bounds
- `src/tests/schemas/shapes/config/public.spec.ts` — Tests for TTL and passphrase defaults
- `src/tests/schemas/shapes/config/sections.spec.ts` — Spot-checks for section shape defaults
- `src/tests/schemas/contracts/config/public.spec.ts` — Updated to verify contract is type-only (no defaults)

### Design Rationale

The contract/shape split eliminates duplication: previously, adding a default to a nested field required restating the entire parent object. Now, `augment` walks the contract tree and applies overrides only where specified. This also clarifies intent — contracts are the source of truth for field names and types; shapes are where runtime behavior (defaults, bounds) lives.

## Test Plan

Added comprehensive unit tests for the `augment` helper (231 lines) covering leaf transforms, sub-tree recursion, wrapper handling, and edge cases. Added smoke tests for billing and public shapes to verify bounds are restored. Updated contract-level tests to confirm contracts remain type-only. Existing JSON Schema generation round-trip tests exercise the composed shapes end-to-end.

https://claude.ai/code/session_01LqjPZdGzvKM5qNUtd6GTDZ

### CODE REVIEW

File: src/schemas/utils/augment.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+      nextShape[key] = rewrap(augment(inner, override));
+    }
+  }
+
+  let next: any = z.object(nextShape);
+  const catchall = (contract as any).def?.catchall;
+  if (catchall) {
+    next = next.catchall(catchall);
+  }
+  return next as T;
```
[the remark this answers is not in the document]
Comment (delano) [reply src/schemas/utils/augment.ts:135]: This suggestion is based on Zod v3 semantics. In Zod v4 there is no `def.unknownKeys` — all unknown-key behavior is encoded via `def.catchall`, which the helper already preserves:

| Source form | `def.catchall.def.type` |
| --- | --- |
| `z.strictObject(...)` | `never` |
| `z.looseObject(...)` | `unknown` |
| `z.object(...).strict()` | `never` |
| `z.object(...).passthrough()` | `unknown` |
| `z.object(...)` (plain) | undefined → strip (default) |

`augment` reads `contract.def.catchall` and re-applies it after `z.object(nextShape)`, so all four behaviors round-trip. The existing `z.strictObject` test covered the strict case; I added explicit tests for `.strict()`, `.passthrough()`, and the strip default in ebd4989 so the coverage is exhaustive.

---
_Generated by [Claude Code](https://claude.ai/code)_
